server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -79,6 +92,10 @@ public ContextIndexSearcher(IndexReader reader, Similarity similarity, QueryCach
     }
 
     public void setProfiler(QueryProfiler profiler) {
+        if (profiler != null) {
+            // disable query caching on profiled query
+            setQueryCachingPolicy(NEVER_CACHE_POLICY);
+        }
```

Comment:
out of curiosity, why isn't returning false on ProfileWeight enough?

Reply:
Because the sub-query might be cacheable so returning `false` only ensures that we don't cache the query at the profile weight level. Sub queries are still eligible for caching so we have to disable caching entirely.

Reply:
I thought we wrapped sub weights too?

Reply:
We do and that works for intermediate level but leaves are still eligible for caching so a simple range query could get cached ?

Reply:
Wouldn't the range weight be wrapped as well?